nifi-commons/nifi-web-client/src/main/java/org/apache/nifi/web/client/StandardWebClientService.java

```diff
@@ -338,7 +346,7 @@ public HttpResponseEntity retrieve() {
             final InputStream responseBody = response.body();
             final InputStream body = responseBody == null ? new ByteArrayInputStream(EMPTY_BYTES) : responseBody;
 
-            return new StandardHttpResponseEntity(code, headers, body);
+            return new StandardHttpResponseEntity(code, headers, body, request.uri());
```

Comment:
It seems like the `response.uri()` method should used here instead.
```suggestion
            return new StandardHttpResponseEntity(code, headers, body, response.uri());
```